website/mails/mail_callbacks.py

```diff
@@ -0,0 +1,57 @@
+# -*- coding: utf-8 -*-
+from datetime import datetime, timedelta
+
+from website import settings
+from modularodm import Q
+from modularodm.exceptions import NoResultsFound, MultipleResultsFound
+
+def no_addon(email):
+    return len(email.user.get_addons()) == 0 and email.user.is_registered
+
+def no_login(email):
+    return email.user.is_registered and not email.user.date_last_login > datetime.utcnow() - settings.NO_LOGIN_WAIT_TIME
+
+def new_public_project(email):
+    """ Will check to make sure the project that triggered this callback is still public
+    before sending the email. It also checks to make sure this is the first (and only)
+    new public project email to be sent
+
+    :param email: QueuedMail object, with 'nid' in its data field
+    :return: boolean based on whether the email should be sent
+    """
+
+    # In line import to prevent circular importing
+    from website.models import Node
+    try:
+        node = Node.find_one(Q('_id', 'eq', email.data['nid']))
```

Comment:
You can get a node by it's id by doing `Node.load(email.data['nid'])`. If it doesn't exist, it will return `None`.